cg/meta/workflow/mutant/quality_controller/report_generator_utils.py

```diff
@@ -0,0 +1,43 @@
+from pathlib import Path
+
+from cg.io.json import write_json
+from cg.meta.workflow.mutant.constants import QUALITY_REPORT_FILE_NAME
+from cg.meta.workflow.mutant.quality_controller.models import (
+    CaseQualityResult,
+    SamplesQualityResults,
+)
+from cg.meta.workflow.mutant.quality_controller.result_logger_utils import samples_results_message
+
+
+def write_report(
+    case_path: Path,
+    case_quality_result: CaseQualityResult,
+    samples_quality_results: SamplesQualityResults,
+) -> None:
+    summary: str = get_summary(
+        case_quality_result=case_quality_result, samples_quality_results=samples_quality_results
+    )
+    report_content = {
+        "summary": summary,
+        "case": case_quality_result.model_dump(),
+        "samples": samples_quality_results.model_dump(),
+    }
```

Comment:
```suggestion
    report_content = ReportContent(summary=summary, case=case_quality_result, samples=samples_quality_results)
```

Then add a model:

```Python
ReportContent(BaseModel):
summary: str
case: CaseQualityResult
samples: SamplesQualityResults
```

Reply:
I think this is what @ivadym was after?

Reply:
This change is already in place in the latest commit. I think you look at an outdated version of the code! 😅